tracer.go

```diff
@@ -0,0 +1,83 @@
+package lumigotracer
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"reflect"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type tracer struct {
+	provider  *sdktrace.TracerProvider
+	logger    logrus.FieldLogger
+	span      trace.Span
+	eventData []byte
+	ctx       context.Context
+	traceCtx  context.Context
+}
+
+func NewTracer(ctx context.Context, cfg Config, payload json.RawMessage) (retTracer *tracer, err error) {
+	defer recoverWithLogs()
+	retTracer = &tracer{
+		ctx:    ctx,
+		logger: logger,
+	}
+
+	exporter, err := createExporter(cfg.PrintStdout, ctx, logger)
+	if err != nil {
+		return nil, errors.Wrap(err, "faialed to create otel exporter")
+	}
+
+	data, err := json.Marshal(&payload)
+	if err != nil {
+		return nil, errors.Wrap(err, "faialed to parse event payload")
+	}
+	retTracer.eventData = data
+
+	tracerProvider := sdktrace.NewTracerProvider(
+		sdktrace.WithSyncer(exporter),
+		sdktrace.WithResource(newResource(ctx,
+			attribute.String("event", string(retTracer.eventData)),
+		)),
+	)
+	retTracer.provider = tracerProvider
+	otel.SetTracerProvider(tracerProvider)
+	panic("skata")
```

Comment:
What does it mean? Why do we panic here?

Reply:
hahaha actually that was testing. removing